### PR TITLE
Solution to placement of saml_config.php

### PR DESCRIPTION
**Pull request by henrikthorn**

### CODE REVIEW

File: auth/saml/auth.php [hunk 1]
```diff
@@ -301,15 +301,14 @@ function process_config($config) {
 	    }
 
 	    if(isset($config->initialize_roles)) {  
-	        global $CFG;
```
Comment (pitbulk) [inline auth/saml/auth.php]: Why remove the global $CFG; ? is required down to get the $CFG->wwwroot

Comment (henrikthorn) [inline auth/saml/auth.php]: Yes, but you get the global $CFG in line 286.

File: auth/saml/index.php [hunk 2]
```diff
@@ -9,11 +9,25 @@
...
+				$config = file_get_contents('../../config.php');
+
+				$config = preg_replace('/\s+/', '', $config);
+				if(preg_match('/^.+?\$CFG\-\>dataroot=[\'"](.+?)[\'"];/', $config, $match)){
+					$CFG->dataroot = $match[1];
+				}else {
+            throw(new Exception('Moodle dataroot not found'));
+				}
+				
+				die(var_dump($CFG));
```
Comment (pitbulk) [inline auth/saml/index.php]: This should be removed? --> die(var_dump($CFG));

File: auth/saml/index.php [hunk 3]
```diff
@@ -9,11 +9,25 @@
         // We read saml parameters from a config file instead from the database
         // due we can not operate with the moodle database without load all
         // moodle session issue.
-        if(!file_exists('saml_config.php')) {
+				//We need to get dataroot from CFG without loading moodle database
+
+				$config = file_get_contents('../../config.php');
+
+				$config = preg_replace('/\s+/', '', $config);
+				if(preg_match('/^.+?\$CFG\-\>dataroot=[\'"](.+?)[\'"];/', $config, $match)){
```
Comment (pitbulk) [inline auth/saml/index.php]: First time this file does not exists, why not create it with default values instead? 

Also in order to apply this patch I need that old version of the plugin keep working, so if someone already saved the file on the saml plugin folder, we need to read the content and save it in the moodledata folder.

File: auth/saml/index.php [hunk 4]
```diff
@@ -9,11 +9,25 @@
...
+					$CFG->dataroot = $match[1];
+				}else {
+            throw(new Exception('Moodle dataroot not found'));
+				}
+				
+				die(var_dump($CFG));
+ 
+        if(!file_exists($CFG->dataroot.'/saml_config.php')) {
```
Comment (pitbulk) [inline auth/saml/index.php]: First time this file does not exists, why not create it with default values instead?

Also in order to apply this patch I need that old version of the plugin keep working, so if someone already saved the file on the saml plugin folder, we need to read the content and save it in the moodledata folder.

File: auth/saml/auth.php (hunk 1 above)
Comment (henrikthorn) [inline auth/saml/auth.php]: I've added a new version without conflicts.

File: auth/saml/index.php (hunk 4 above)
Comment (henrikthorn) [inline auth/saml/index.php]: The file is actually included in git, which should make it impossible to not have the file. However, it should be created during the upgrade/install-process. I can create an upgrade path from a specified version before the file was included, which checks for a present file in the plugin-dir and the copies it to the moodledata-dir, if it has been changed. Agree?

**Comment by pitbulk** [comment]
Adding a upgrade/install-process make sense.

This PR replace the folder of the saml_config.php file. The upgrade process should
review if exists a $CFG->dataroot.'/saml_config.php'. 
a) if not exists, copy the saml_config.php file from the saml plugin folder there.
b) If exists do nothing.

**Comment by hcpss-banderson** [comment]
Picked this up at #27

**Comment by pitbulk** [comment]
I solved this issue on the new release.

**Pull request closed without merging**